.github/workflows/ci.yml

```diff
@@ -0,0 +1,153 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - master
+      - develop
+      - "release/**"
+env:
+  APPLITOOLS_API_KEY: ${{ secrets.APPLITOOLS_API_KEY }}
+  APPLITOOLS_BATCH_ID: ${{ github.sha }}
+  APPLITOOLS_BATCH_NAME: ${{ github.head_ref }}
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: v12.20.0
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: |
+            **/node_modules
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: lerna prepare
+        if: steps.yarn-cache.outputs.cache-hit == 'true'
+        run: lerna run prepare
```

Comment:
this won't check that the lock file does not contain changes that were not checked in. We might consider simplifying here to always run `yarn install --frozen-lockfile --prefer-offline`. Yarn is generally fast if the cache provides the FS state for `**/node_modules`

Reply:
we are already running `yarn install --frozen-lockfile --prefer-offline`, are you suggesting removing the lerna prepare step?

Reply:
we are only running that install command if the cache is cold. My point is that someone can open a PR to get a cache built, then in a subsequent commit change a package.json file, not commit the `yarn.lock` changes and still get a green build because the cache being present will skip the `--frozen-lockfile` check. We need something that always checks that uncommitted yarn.lock changes aren't getting through. You can remove the `if` conditionals here, and the entire `lerna prepare` step and instead just run the `install dependencies` step *even* if we have a warm cache